internal/controller/dispatcher.go

```diff
@@ -143,10 +144,20 @@ func (r *Dispatcher) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		switch appWrapper.Status.Step {
 		case mcadv1beta1.Dispatching:
 			if r.MultiClusterMode {
+				// Serialize current status to an annotation so it can be mirrored in the execution cluster
+				pickledStatus, err := json.Marshal(appWrapper.Status)
+				if err != nil {
+					return ctrl.Result{}, err
```

Comment:
A general comment, not urgent and not really specific to this PR or blocking the PR: how should we capture that this should really never fail?

Reply:
Good question (and similar comes up in the runner in the deserializing code).  Really the only failure that is plausible here is some kind of OOM since the Status has to be well-formed.